Userland/Libraries/LibWeb/XHR/FormData.cpp

```diff
@@ -91,67 +78,68 @@ WebIDL::ExceptionOr<void> FormData::append_impl(String const& name, Variant<JS::
     // 2. Let entry be the result of creating an entry with name, value, and filename if given.
     auto entry = TRY(HTML::create_entry(realm, name, value, filename));
 
-    // FIXME: Remove this when our binding generator supports "new string".
-    auto form_data_entry_value = entry.value.has<String>()
-        ? FormDataEntryValue { entry.value.get<String>().to_deprecated_string() }
-        : FormDataEntryValue { entry.value.get<JS::NonnullGCPtr<FileAPI::File>>() };
-
     // 3. Append entry to this’s entry list.
-    if (auto entries = m_entry_list.get(entry.name.to_deprecated_string()); entries.has_value() && !entries->is_empty())
-        TRY_OR_THROW_OOM(vm, entries->try_append(form_data_entry_value));
-    else
-        TRY_OR_THROW_OOM(vm, m_entry_list.try_set(entry.name.to_deprecated_string(), { form_data_entry_value }));
-
+    TRY_OR_THROW_OOM(vm, m_entry_list.try_append(move(entry)));
     return {};
 }
 
 // https://xhr.spec.whatwg.org/#dom-formdata-delete
-void FormData::delete_(DeprecatedString const& name)
+WebIDL::ExceptionOr<void> FormData::delete_(String const& name)
 {
     // The delete(name) method steps are to remove all entries whose name is name from this’s entry list.
-    m_entry_list.remove(name);
+    m_entry_list.remove_first_matching([&name](FormDataEntry const& entry) {
+        return entry.name == name;
+    });
+    return {};
 }
 
 // https://xhr.spec.whatwg.org/#dom-formdata-get
-Variant<JS::NonnullGCPtr<FileAPI::File>, DeprecatedString, Empty> FormData::get(DeprecatedString const& name)
+WebIDL::ExceptionOr<Variant<JS::Handle<FileAPI::File>, String, Empty>> FormData::get(String const& name)
```

Comment:
This function never returns an error. Maybe there are reasons for having it return `WebIDL::ExceptionOr`?

Reply:
Oops, I forgot to remove this when I removed the DeprecatedString -> String conversion inside this function _before_ I added UseNewAKString.